server/update/middleware.ts

```diff
@@ -228,6 +229,33 @@ import UpdateCollection from '../update/collection';
     }
   }
 
+  // tags, if defined, must be a list of strings corresponding to tags in the project
+  if (tags !== undefined) {
+    if (!Array.isArray(tags)) {
+      res.status(401).json({
+        error: 'Tags must be a list.'
+      });
+      return;
+    }
+
+    const project = await ProjectCollection.findOne(req.body.projectId);
+    const projectTags = project.tags;
+
+    for (const tag of tags) {
+      if (typeof tag !== 'string') {
+        res.status(401).json({
+          error: 'Tags must be a list of strings.'
+        });
+        return;
+      } else if (!projectTags.includes(tag)) {
```

Comment:
same suggestion here as prev comment

Reply:
I decided to remove this completely and only do the projectTags thing—it's got to be a valid tag if it's in projectTags, so it should be enough just to check that it's in there!